src/views/LayoutWithRouting.tsx

```diff
@@ -44,6 +53,21 @@ const Route: React.FC<RouteProps> = ({ Component, ...pathProps }) => {
   )
 }
 
+type RoutesProps = Omit<RouterProps, 'component'>
+const Routes: React.FC<RoutesProps> = (props) => (
+  <Router {...props}>
+    <Route default Component={HomeView} />
+    <Route path={routes.video()} Component={VideoView} />
+    <Route path={routes.searchOverlay()} Component={SearchOverlay} />
```

Comment:
This doesn't seem to be working as expected. The search overlay should exist on top of the currently selected view. That means that both the overlay and previous route should be displayed at the same time. The example desired behaviour is this:
1. I open Videos view, pick a category, scroll a bit
2. I open the search overlay
3. I close the search overlay
4. I'm left with the exact state as it was after step 1 - Videos view shouldn't get unmounted&mounted in the process